packages/oui-file/src/file.html

```diff
@@ -0,0 +1,132 @@
+<!-- Controls -->
+<input type="hidden"
+    ng-attr-id="{{::$ctrl.id}}"
+    ng-attr-name="{{::$ctrl.name}}"
+    ng-disabled="$ctrl.disabled"
+    ng-required="$ctrl.required"
+    ng-model="$ctrl.model">
+<input class="oui-file__input" type="file"
+    ng-if="::!$ctrl.attachments"
+    ng-attr-id="{{::$ctrl.selectorId}}"
+    ng-attr-accept="{{::$ctrl.accept}}"
+    ng-attr-size="{{::$ctrl.size}}"
+    ng-disabled="$ctrl.disabled"
+    tabindex="-1">
+<input class="oui-file__input" type="file"
+    ng-if="::$ctrl.attachments"
+    ng-attr-id="{{::$ctrl.selectorId}}"
+    ng-attr-accept="{{::$ctrl.accept}}"
+    ng-attr-size="{{::$ctrl.size}}"
+    ng-disabled="$ctrl.disabled"
+    tabindex="-1"
+    multiple>
+<!-- /Controls -->
+
+<!-- Selector -->
+<div class="oui-file-selector"
+    ng-if="::!$ctrl.multiple && !$ctrl.droparea"
+    ng-class="{
+        'oui-file-selector_error': $ctrl.form && $ctrl.form[$ctrl.name].$invalid && ($ctrl.form[$ctrl.name].$touched || $ctrl.form.$submitted)
+    }">
+
+    <!-- Fake Selector -->
+    <div class="oui-file-selector__input">
+        <button class="oui-file-selector__label" type="button"
+            ng-blur="$ctrl.setInputTouched()"
```

Comment:
Set the input as touched when it's blured ?

Reply:
Form validation